CMakeLists.txt

```diff
@@ -1475,19 +1462,17 @@ else()
 endif()
 add_dependencies(mixxx-lib mixxx-res mixxx-script)
 
+file(READ src/_version.h MIXXX_VERSION_FILECONTENT)
+string(REGEX REPLACE "^.*#define MIXXX_VERSION \"(.*)\".*$" "\\1" MIXXX_VERSION "${MIXXX_VERSION_FILECONTENT}")
+
```

Comment:
MXXX_VERSION must match the file Tag. Can we check that and fail if it is not the case. ? 
Or add a big warning and continue with the tag version. This is at least consistent to what is shown in the about box.

Reply:
Why do we even need to read it from the file? Can't we just use VERSION from the cmake project call?

Reply:
Yes I think we should get rid of the file.

Reply:
Unfortunately CMake has no concept of a prerelease version. With `project(mixxx VERSION 2.3-beta)`, cmake fails:

```
CMake Error at CMakeLists.txt:2 (project):
  VERSION "2.3-beta" format invalid.
```

Reply:
Because CMake's VERSION has no way to specify prerelease versions, I think we should keep the _version.h file and not rely on the CMake VERSION.

Reply:
> MXXX_VERSION must match the file Tag. Can we check that and fail if it is not the case. ?

I think we should not do this. We should only make the tag after a final release commit has been QA tested on every OS. Now that the version comes from Git, this will require making the tag after QA then rebuilding the same commit. So the _version.h file should be updated before making the tag.

Reply:
Ah OK.  That sound complicated. 
I think the normal way would be to create the release binary an tag the commit afterwards. 
This means we have a gap where the release is build but we have not yet tagged the commit. 
During that time the file _version.h and the git version are different. 

I think we should give the file version priority. 

How about special case that and using a predicted GIT_COMMIT_DESCRIPTION like this: 
2.3.0-0-g123124

The nice thing is that the tag commit would be 2.3.0-1-g123124, so we can distinguished the golden release binary from later builds.

Reply:
>  I think the normal way would be to create the release binary an tag the commit afterwards.

Yes, that is what we have done in the past.

> I think we should give the file version priority... How about special case that

I think that would be overcomplicated. Currently it is relatively simple. If CMake is run in a Git repository, the version comes from `git describe`; otherwise, it comes from _version.h.

Reply:
Dealing with a second build after pushing a tag is over complicated. The extra code here is only a one time issue. 
The Rule is ease: 
The version comes always form _version.h. A release build must be tagged with the same version to enable automatic build number.

Reply:
> The version comes always form _version.h

That would defeat the purpose of this PR. If we had some complex string manipulation to generate something like the `git describe` output there is no guarantee that it would actually work when used with git commands.

Reply:
Honestly I don't see the issue. We just push the final commit, wait for the build to finish, check if the binaries are okay, the tag that exact commit push it. It's not more work or something like that.

Reply:
Ok, so we need a real release commit including the tag. Just pushing the tag will not work.
That can be supprising, and does not match the GitHub realease workflow.

I think we are back to the original proposal to assert that the file version match. 
If we use always the tag version, we have no check at all, that the _version.h is up to date.

What are your ideas to archive this?